.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,29 @@
+name: build-and-test
+
+on:
+  push:
+  pull_request:
```

Comment:
While we are enabling GH Actions, can we set them to run nightly as well?

Reply:
we can, you mean to ensure that it works with the latest snapshots?

Reply:
Yeah, that was my goal.  I suppose we'd have to both have a nightly GH Action and make sure to increment the GM version in the pom to the nightly GM snapshot version.

Does that seem reasonable?

Reply:
lets add that as a follow on task if you don't mind... we probably want to think through it.

Reply:
Works for me.  I've entered a ticket here: https://geomesa.atlassian.net/browse/GEOMESA-3029 and tagged it for 4.0.0.